architecture/trusted_applications.rst

```diff
@@ -76,13 +76,25 @@ the suffix ``.ta``. They are built separately from the OP-TEE core boot-time
 blob, although when they are built they use the same build system, and are
 signed with the key from the build of the original OP-TEE core blob.
```

Comment:
"[REE filesystem TA] consist of a cleartext signed ELF file": this is no longer true (not cleartext).
I think the paragraph "REE filesystem TAs come in three formats" below should be first (here) in order to introduce the three types. That is: explain TAs are ELF files, signed and optionally encrypted with `scripts/sign_encrypt.py` to produce `.ta` files, and come in three types.
1. Legacy TAs, signed, not encrypted, cannot be created anymore by the build scripts since version 3.7.0, no need to explain much more I think
2. Bootstrap TAs, signed with the key from the build of the original OP-TEE core blob, not encrypted
3. Encrypted TAs (should we say "encrypted bootstrap TAs"?), sign-then-encrypt-then-MAC, encrypted with `TA_ENC_KEY` when `CFG_ENCRYPT_TA=y`, and explain the `tee_otp_get_ta_enc_key()` thing.

Reply:
Okay, will re-structure sections as per your suggestions.

> should we say "encrypted bootstrap TAs"?

I guess via bootstrap TAs, we meant to use them only once to install TAs in Secure Storage which will be used to load TAs later on. But with pre-encrypted TAs in REE-FS, user may not want to install them in Secure Storage in which case we won't call them as bootstrap TAs as OP-TEE will load encrypted TAs from REE-FS only. Although, there maybe use-cases where installing encrypted TAs in Secure Storage may be useful.

So I would prefer to just call them as encrypted TAs. As it depend on users whether they would like to use them as bootstrap TAs installed in Secure Storage or TAs loaded from REE-FS.